cosim/black-parrot-example/fpga/Makefile

```diff
@@ -0,0 +1,79 @@
+.PHONY: fpga_build
+.PHONY: load_bitstream
+
+# update this with a pointer to your vivado install
+XILINX_ENV=source /tools/Xilinx/Vivado/2019.1/settings64.sh
+
+# need to set the shell so that source works
+SHELL := /bin/bash
+
+PS_FILE=../ps.cpp
+BASENAME=black_parrot
+NBF_FILE=../prog.nbf
+
+# you must set these for the accelerator to work
+
+CFLAGS=-DGP0_ADDR_BASE=0x40000000 -DGP0_ADDR_SIZE_BYTES=4096 -DGP1_ADDR_BASE=0x80000000 -DGP1_ADDR_SIZE_BYTES=0x30000000 -DFPGA
+
+all:
+	@echo See the makefile for the fun things you can do in this directory
+
+# you do this on the zynq board itself
+run:	load_bitstream
+	g++ -I../../include/fpga/ $(CFLAGS) -O2 $(PS_FILE) -lcma -lpthread
+	sudo ./a.out $(NBF_FILE)
+
+fpga_build:
+	$(MAKE) fpga_build_ip
+	$(MAKE) fpga_build_block
+
+# do this on a server that has vivado 2019.X
+# package BP IP, and then synthesize, pnr and bitfile generate
+fpga_build_block:
+	$(XILINX_ENV) && vivado -source vivado-create-block.tcl -mode tcl
+
+# this packages the BP IP, but does not continue to synthesis, pnr and bitfile generation
+fpga_build_ip: clean_fpga_build flist.vcs
+	mkdir fpga_build
+	$(XILINX_ENV) && vivado -source vivado-build-ip.tcl -mode tcl
```

Comment:
Is there a reason to use -mode tcl over -mode batch? The difference is that batch will terminate the script after whereas tcl breaks to shell, but it seems like we don't really want that

Reply:
yes, to allow the user to pull up in vivado and inspect.